src/api/component.js

```diff
@@ -283,37 +283,70 @@ export default class extends HTMLElement {
       }
     }
 
-    if (attributeChanged) {
+    // DEPRECATED
+    //
+    // static attributeChanged()
+    const { attributeChanged } = this.constructor;
+    if (isFunction(attributeChanged)) {
       attributeChanged(this, { name, newValue, oldValue });
     }
   }
 
   // Skate
   //
-  // Maps to the static updated() callback. That logic should be moved here
-  // when that is finally removed.
-  updatedCallback (prev) {
-    return this.constructor.updated(this, prev);
+  updatedCallback (prevProps) {
+    // DEPRECATED
+    //
+    // static updated()
+    const { updated } = this.constructor;
+    if (isFunction(updated)) {
+      return updated(this, prevProps);
+    }
+
+    // updatedCallback() default implementation
```

Comment:
Maybe better to say "default implementation short-circuits if this is the first render"?